File: Stack.c
#include<stdio.h>
#include<cs50.h>
int a[10];
int top = -1;
// inserting elements in stack 
void  push(int x){
    if (top==10){
        printf("overflow");
        return;
    }
    top+=1;
    a[top] = x;
} 
deleting element in a stack 
void pop(){
    if (top==-1){
        printf("underflow");
        return;
    }
    top--;
}
int Top(){
    return a[top];
}
/*void  isempty(){
    if (top==-1){
        return true;
    }
    return false;
}*/
// print elements in stack // 
void Print(){
    for(int i =0;i<=top;i++){
        printf("%i",a[i]);
    }
    printf("\n");
}
int main (){
    push(1);
    push(2);
    push(3);
   // pop();
    Top();
    Print();
    push(4);
    Print();

}

File: Stack.java

public class stack{
    int stack[] = new int [5];
    int top = 0;
    public void push(int data){
        stack[top] = data;
        top ++;
    }
    public void print(){
        for (int n :stack){
            System.out.println(n+"");
        }
    }
    public void pop(){
        top --;
        stack[top] = 0;

    }
    public static void main(String args[]){
        stack obj = new stack();
        obj.push(1);
        obj.push(2);
        obj.push(3);
        obj.push(4);
        obj.pop();
        obj.print();
    }
}
